src/app.rs

```diff
@@ -2,12 +2,98 @@ use crate::profile::Profile;
 use crate::profile_list_item::ProfileListItem;
 use candidate::Candidate;
 use std::collections::HashMap;
+use std::collections::HashSet;
 use yew::prelude::*;
 use yew_router::{router::Router, Switch};
 use yewprint::{Button, IconName, InputGroup, Text, H1, H2};
 
 pub struct App {
-    candidates: HashMap<&'static str, &'static Candidate>,
+    candidates: HashMap<&'static str, CandidateInfo>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Tech {
+    pub tech: &'static str,
+    pub professional: bool,
+    pub public: bool,
+}
+
+impl Tech {
+    fn from_tech(candidate_tech: &'static str) -> Tech {
+        let tech = candidate_tech;
+        let professional = false;
+        let public = false;
+        Tech {
+            tech,
+            professional,
+            public,
+        }
+    }
+    fn from_pro_tech(candidate_tech: &'static str) -> Tech {
+        let tech = candidate_tech;
+        let professional = true;
+        let public = false;
+        Tech {
+            tech,
+            professional,
+            public,
+        }
+    }
+    fn from_pub_tech(candidate_tech: &'static str) -> Tech {
+        let tech = candidate_tech;
+        let professional = false;
+        let public = true;
+        Tech {
+            tech,
+            professional,
+            public,
+        }
+    }
```

Comment:
```suggestion
    fn from_pub_tech(tech: &'static str) -> Tech {
        Tech {
            tech,
            professional: false,
            public: true,
        }
    }
```
No need to be so verbose